.travis.yml

```diff
@@ -1,23 +1,80 @@
+  - CCX=gcc-4.8 FULL_CHECK=true # osx
+  - CXX=g++-4.8 FULL_CHECK=false # linux
+  - CXX=g++-4.8 FULL_CHECK=true # linux
+  - global:
+      secure: "EuQQSD9GKeAUtFhNGe/m23IYXKVTr8CPl9f7O/PsVDkDMsO6E15o16hn9rn/kIeMEE9+UsCGapFJYlTg+jvZncr8kNTWkKTxsaYS9ejc9br4ePzJmg4LXrelIsunkUGTRBG7wd5I7g+Tx24k02EItftYhzhw3GV3sDvw7VISts3CaCFrGMwOlI4vuhFxJ7N9XRg3HFPY4O9847R/nfhbN9eMqI1kg9xA2INoYaeqxxjWjCX/YlRNpUjkpauv3Owc9Z+B5DYSAlywTModqiUdkiAGmFptydo8gkCqll+iuFfZAV7KT338gVVP3XLNVP1Bq7ZrcqPimhepStX1smEo7vtPzUSb4Wiy03hXlkhDEhLzOf629bGrwpMDy49oAYN0+7o2JBmtV91VEOOs2+ziDTprdv3l4VyX9em+/tf3KxZqTNuiq2Rbl2607zQ3MVYzvHqedoAJiimzUAqMEC4JZmAhlS4dnLFpG2aZv0vuNS9hzVcmSrwcUUwQXWDuRVJkI2mpvL/avbSSH2+kww+I2Uz6VPIOnWoMlY+PJ1i7nMGLpbRY9D4QY1JGJh897hSTHw3F21/4h7kSrkPG0rIdCTCiJ3sXBfUdeSdBi0E+ca9vUWZBbsfH0NdRcw4n0cYiG0NaV1zNA6EIRi8L1GHLtPtYbu/Gm2HM63d7agIQvR0="
+cache:
+  directories:
+    - "node_modules"
 matrix:
+  # Don't build for incompatible c compilers.
+  exclude:
+    - os: osx
+      env: CXX=g++-4.8 FULL_CHECK=false
+    - os: osx
+      env: CXX=g++-4.8 FULL_CHECK=true
+    - os: linux
+      env: CCX=gcc-4.8 FULL_CHECK=false
+    - os: linux
+      env: CCX=gcc-4.8 FULL_CHECK=true
+  # Allow failures for all included runs where FULL_CHECK=true (linting)
   allow_failures:
-    - env: FULL_CHECK=true
+    - os: osx
+      env: CCX=gcc-4.8 FULL_CHECK=true
+    - os: linux
+      env: CXX=g++-4.8 FULL_CHECK=true
+# before_install:
 before_script:
-  - npm install -g license-checker
+  # `npm install` is automatic
+  - npm install -g node-gyp license-checker
+  # Install cargo and then `cargo install emerald` and move binary to project base.
+  - bash ./dependencies.sh
+  # Set up GCP
+  - npm install gcloud-storage-upload
+  - openssl aes-256-cbc -k "$GCP_PASSWD" -in gcloud-travis.json.enc -out .gcloud.json -d
 script:
+  # Generate current version information from git tags.
+  - git describe --tags --always > version.txt
+  - sed -E 's/v([[:digit:]]+\.[[:digit:]]+)\.[[:digit:]]-([[:digit:]]+)-g([a-f0-9]+)/v\1.\2+\3/' version.txt > version-app.txt
+  - sed -E 's/v([[:digit:]]+\.[[:digit:]]+)\.[[:digit:]]-([[:digit:]]+).+/v\1.\2/' version.txt > version-only.txt
+  - sed -E 's/v([[:digit:]]+\.[[:digit:]]+)\.[[:digit:]]-([[:digit:]]+).+/v\1.x/' version.txt > version-base.txt
+  # We can use this env variable in electron-builder for naming, and avoid the `for ext in ...` bash loop below.
+  - export APP_VERSION_GIT_TAG="$(cat version-app.txt)"
+  - echo "Building app version $APP_VERSION_GIT_TAG"
   - if [[ $(license-checker --production --exclude '(GPL-2.0 OR MIT)' | grep GPL) ]]; then license-checker --production | grep -C 1 GPL && false; fi
   - npm run build
   - if [[ $FULL_CHECK == "true" ]]; then npm run lint --silent; fi
-env:
-  - CXX=g++-4.8
+  # Don't sign the build for now.
+  # Don't build releases for lint tests.
+  - if [[ ! "$FULL_CHECK" ]]; then env CSC_IDENTITY_AUTO_DISCOVERY=false npm run dist && ls -l ./dist/; fi
+  # Build dist
+  - npm run dist
```

Comment:
We should either get rid of the `$FULL_CHECK` conditional above (which is intended to not run duplicate builds for CI builds testing against linters... a statement which may or may not not be working properly since I haven't looked into it) or get rid of line 62. They're doing the same thing. 

If we keep line 62, we may need to add the `CSC_IDENTITY_AUTO_DISCOVERY=true` var to disable signing for osx.